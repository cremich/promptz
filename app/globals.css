pre,
code {
  word-wrap: break-word !important;
  white-space: pre-wrap !important;
  white-space: -moz-pre-wrap !important;
  white-space: -pre-wrap !important;
  white-space: -o-pre-wrap !important;
  margin: 0;
  padding: 0;
}

code {
  background-color: #f3f3f7;
}

pre > code {
  display: inline-block;
  padding: 1em;
}
