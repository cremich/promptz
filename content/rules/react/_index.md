---
title: "React"
description: "Rules for React development, component patterns, and best practices."
---
