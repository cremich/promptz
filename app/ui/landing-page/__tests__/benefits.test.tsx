import { describe, expect, test } from "@jest/globals";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import Benefits from "@/app/ui/landing-page/benefits";

describe("Benefits", () => {
  test("renders benefits unchanged", () => {
    const { container } = render(<Benefits />);
    expect(container).toMatchSnapshot();
  });
});
