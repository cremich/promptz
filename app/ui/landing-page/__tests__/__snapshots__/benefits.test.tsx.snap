// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Benefits renders benefits unchanged 1`] = `
<div>
  <section
    class="w-full"
  >
    <div
      class="px-4 md:px-6"
    >
      <div
        class="flex flex-col items-center justify-center space-y-4 text-center"
      >
        <div
          class="space-y-2"
        >
          <h2
            class="text-3xl font-bold tracking-tighter sm:text-5xl"
          >
            Why Promptz?
          </h2>
          <p
            class="max-w-[600px] text-muted-foreground md:text-xl/relaxed lg:text-base/relaxed xl:text-xl/relaxed"
          >
            The perfect prompt is just one click away!
          </p>
        </div>
      </div>
      <p
        class="mt-4 text-center max-w-4xl mx-auto text-zinc-400 mb-12"
      >
        PROMPTZ is your ultimate resource for Amazon Q Developer prompt engineering. Discover, create, and share high-quality prompts to tackle real-world software development challenges. From generating AWS architecture blueprints to automating workflows and beyond, PROMPTZ empowers developers to get the most out of Amazon Q Developer with a rich library of prompt templates and community contributions.
      </p>
      <div
        class="mt-12 grid gap-8 md:grid-cols-2 lg:grid-cols-3"
      >
        <div
          class="rounded-lg border bg-neutral-800 text-card-foreground shadow-sm text-center"
        >
          <div
            class="flex flex-col space-y-1.5 p-6"
          >
            <div
              class="text-2xl font-semibold leading-none tracking-tight items-center flex flex-col"
            >
              <svg
                class="lucide lucide-code w-12 h-12 text-violet-500 mb-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="16 18 22 12 16 6"
                />
                <polyline
                  points="8 6 2 12 8 18"
                />
              </svg>
              Ready-to-Use Solutions
            </div>
          </div>
          <div
            class="p-6 pt-0 text-zinc-400"
          >
             Simplify software development with our curated collection of production-ready prompt templates
          </div>
        </div>
        <div
          class="rounded-lg border bg-neutral-800 text-card-foreground shadow-sm text-center"
        >
          <div
            class="flex flex-col space-y-1.5 p-6"
          >
            <div
              class="text-2xl font-semibold leading-none tracking-tight items-center flex flex-col"
            >
              <svg
                class="lucide lucide-users w-12 h-12 text-violet-500 mb-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16 21v-2a4 4 0 0 0-4-4H6a4 4 0 0 0-4 4v2"
                />
                <circle
                  cx="9"
                  cy="7"
                  r="4"
                />
                <path
                  d="M22 21v-2a4 4 0 0 0-3-3.87"
                />
                <path
                  d="M16 3.13a4 4 0 0 1 0 7.75"
                />
              </svg>
              Community Collaboration
            </div>
          </div>
          <div
            class="p-6 pt-0 text-zinc-400"
          >
            Join a vibrant community to refine and share your prompt engineering ideas
          </div>
        </div>
        <div
          class="rounded-lg border bg-neutral-800 text-card-foreground shadow-sm text-center"
        >
          <div
            class="flex flex-col space-y-1.5 p-6"
          >
            <div
              class="text-2xl font-semibold leading-none tracking-tight items-center flex flex-col"
            >
              <svg
                class="lucide lucide-book-open w-12 h-12 text-violet-500 mb-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 7v14"
                />
                <path
                  d="M3 18a1 1 0 0 1-1-1V4a1 1 0 0 1 1-1h5a4 4 0 0 1 4 4 4 4 0 0 1 4-4h5a1 1 0 0 1 1 1v13a1 1 0 0 1-1 1h-6a3 3 0 0 0-3 3 3 3 0 0 0-3-3z"
                />
              </svg>
              Best Practices
            </div>
          </div>
          <div
            class="p-6 pt-0 text-zinc-400"
          >
            Learn prompt engineering best practices for every stage of the software development lifecycle
          </div>
        </div>
      </div>
      <div
        class="mt-12 max-w-2xl mx-auto"
      >
        <div
          class="rounded-lg border bg-neutral-800 text-card-foreground shadow-sm text-center"
        >
          <div
            class="flex flex-col space-y-1.5 p-6"
          >
            <div
              class="text-2xl font-semibold leading-none tracking-tight items-center flex flex-col"
            >
              <svg
                class="lucide lucide-github w-12 h-12 text-violet-500 mb-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15 22v-4a4.8 4.8 0 0 0-1-3.5c3 0 6-2 6-5.5.08-1.25-.27-2.48-1-3.5.28-1.15.28-2.35 0-3.5 0 0-1 0-3 1.5-2.64-.5-5.36-.5-8 0C6 2 5 2 5 2c-.3 1.15-.3 2.35 0 3.5A5.403 5.403 0 0 0 4 9c0 3.5 3 5.5 6 5.5-.39.49-.68 1.05-.85 1.65-.17.6-.22 1.23-.15 1.85v4"
                />
                <path
                  d="M9 18c-4.51 2-5-2-7-2"
                />
              </svg>
              Contribute to PROMPTZ
            </div>
          </div>
          <div
            class="p-6 pt-0 text-zinc-400"
          >
            PROMPTZ is a community-first platform. The source code is open on GitHub, and we welcome contributions from developers and enthusiasts. If you want to share your ideas, join us on Github.
          </div>
          <div
            class="flex items-center p-6 pt-0 justify-center"
          >
            <a
              class="px-6 py-3 rounded-full bg-violet-600 hover:bg-violet-700 text-white font-medium transition-colors"
              href="https://github.com/cremich/promptz"
            >
              Join on GitHub
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
